pkg/artifactory/resource/security/resource_artifactory_scoped_token.go

```diff
@@ -0,0 +1,285 @@
+	TokenType   string `json:"token_type"`
+}
+
+type AccessTokenGet struct {
+	TokenId     string `json:"token_id"`
+	Subject     string `json:"subject"`
+	Expiry      int    `json:"expiry"`
+	IssuedAt    int    `json:"issued_at"`
+	Issuer      string `json:"issuer"`
+	Description string `json:"description"`
+	Refreshable bool   `json:"refreshable"`
+}
+
+func (a AccessTokenPostResponse) Id() string {
+	return a.TokenId
+}
+
+const accessTokensUrl = "access/api/v1/tokens"
+const accessTokenUrl = accessTokensUrl + "/{id}"
+
+func ResourceArtifactoryScopedToken() *schema.Resource {
+
+	var unpackAccessTokenPostRequest = func(data *schema.ResourceData) (*AccessTokenPostRequest, error) {
+		d := &util.ResourceData{data}
+
+		scopes := d.GetSet("scopes")
+		scopesString := strings.Join(scopes, " ") // Join slice into space-separated string
+		if len(scopesString) > 500 {
+			return nil, fmt.Errorf("Total combined length of scopes field exceeds 500 characters: %s", scopesString)
+		}
+
+		audiences := d.GetSet("audiences")
+		audiencesString := strings.Join(audiences, " ") // Join slice into space-separated string
+		if len(audiencesString) > 255 {
+			return nil, fmt.Errorf("Total combined length of audences field exceeds 255 characters: %s", audiencesString)
+		}
+
+		accessToken := AccessTokenPostRequest{
+			Username:    d.GetString("username", false),
+			Scope:       scopesString,
+			ExpiresIn:   d.GetInt("expires_in", false),
+			Refreshable: d.GetBool("refreshable", false),
+			Description: d.GetString("description", false),
+			Audience:    audiencesString,
+		}
+
+		return &accessToken, nil
+	}
+
+	var packAccessToken = func(d *schema.ResourceData, accessToken AccessTokenGet) diag.Diagnostics {
+		setValue := util.MkLens(d)
+
+		var errors []error
+
+		errors = append(errors, setValue("subject", accessToken.Subject)...)
+		errors = append(errors, setValue("expiry", accessToken.Expiry)...)
+		errors = append(errors, setValue("issued_at", accessToken.IssuedAt)...)
+		errors = append(errors, setValue("issuer", accessToken.Issuer)...)
+		errors = append(errors, setValue("description", accessToken.Description)...)
+		errors = append(errors, setValue("refreshable", accessToken.Refreshable)...)
+
+		if len(errors) > 0 {
+			return diag.Errorf("failed to pack access token %q", errors)
+		}
+
+		return nil
+	}
+
+	var accessTokenRead = func(_ context.Context, data *schema.ResourceData, m interface{}) diag.Diagnostics {
+		accessToken := AccessTokenGet{}
+
+		_, err := m.(*resty.Client).R().
+			SetPathParam("id", data.Id()).
+			SetResult(&accessToken).
+			Get(accessTokenUrl)
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		return packAccessToken(data, accessToken)
+	}
+
+	var packAccessTokenPostResponse = func(d *schema.ResourceData, accessToken AccessTokenPostResponse) diag.Diagnostics {
+		setValue := util.MkLens(d)
+
+		var errors []error
+		errors = append(errors, setValue("scopes", strings.Split(accessToken.Scope, " "))...)
```

Comment:
This sort of thing _shouldn't_ be necessary. Arrays are pointers so this should work without this

https://github.com/jfrog/terraform-provider-shared/blob/448604c5362d47a68861b6460c65cb3dd9d9e294/util/util.go#L95-L103